packages/compiler-cli/src/ngtsc/docs/src/class_extractor.ts

```diff
@@ -268,16 +274,43 @@ class ClassExtractor {
   }
 
   /** Gets whether the given signature declaration is documentable. */
-  private isDocumentableSignature(signature: ts.SignatureDeclaration):
-      signature is SignatureElement {
-    return ts.isConstructSignatureDeclaration(signature) ||
-        ts.isCallSignatureDeclaration(signature);
+  private isDocumentableSignature(
+      signature: ts.SignatureDeclaration,
+      ): signature is SignatureElement {
+    return (
+        ts.isConstructSignatureDeclaration(signature) || ts.isCallSignatureDeclaration(signature));
   }
 
   /** Gets whether the declaration for this extractor is abstract. */
   private isAbstract(): boolean {
     const modifiers = this.declaration.modifiers ?? [];
-    return modifiers.some(mod => mod.kind === ts.SyntaxKind.AbstractKeyword);
+    return modifiers.some((mod) => mod.kind === ts.SyntaxKind.AbstractKeyword);
+  }
+
+  /**
+   * Check if the member has a JSDoc @internal or a @internal is a normal comment
+   */
+  private isInternal(member: MethodLike|PropertyLike): boolean {
+    return (
+        extractJsDocTags(member).some((tag) => tag.name === 'internal') ||
+        this.hasLeadingInternalComment(member));
+  }
+
+  /*
+   * Check if the member has a comment block with @internal
+   */
+  private hasLeadingInternalComment(member: MethodLike|PropertyLike): boolean {
+    return (
+        ts.reduceEachLeadingCommentRange(
+            member.parent.getText(),
+            member.getFullStart(),
+            (pos, end, kind, state) => {
+              return member.parent.getText().slice(pos, end).includes('@internal') || state;
```

Comment:
Nit: to not check the text if we already know it's `@internal`.

```suggestion
              return containsInternal || member.parent.getText().slice(pos, end).includes('@internal');
```